### PR TITLE
added prettier configuration with eslint

### PR DESCRIPTION
**Pull request by rohanmittal1163**
## Description 📝
Hey @lalitmee 
Extremely sorry for the delay. 
i have created two files .prettierrc and .eslintrc and added the important options that every project should or must have in order for manual formatting to get the clean code
kindly look to it.
once again sorry for the delay

Fixes/Closes: #193 

## Checklist ✅

- [x] Reviewed the code changes thoroughly.
- [ ] Tested the changes locally and ensured they work as expected.
- [x] Updated relevant documentation.
- [x] Added appropriate test cases, if applicable.
- [ ] Considered potential security or performance implications.
- [x] Code adheres to the project's style guidelines.
- [ ] Addressed all reviewer comments and concerns, if any.

## Screenshots (if applicable)

image
image

## Additional Notes 📒
once again sorry for the delay

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "printWidth": 80,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "trailingComma": "es5",
+  "bracketSpacing": true,
+  "bracketSameLine": true,
```
Comment (lalitmee) [inline .prettierrc:7]:
```suggestion
  "bracketSameLine": false,
```

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "arrowParens": "always",
```
Comment (lalitmee) [inline .prettierrc:8]:
```suggestion
  "arrowParens": "avoid",
```

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  "semi": false,
+  "useTabs": true,
```
Comment (lalitmee) [inline .prettierrc:10]:
```suggestion
  "useTabs": false,
```
Lets use `spaces` instead of `tabs`

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "printWidth": 80,
+  "tabWidth": 2,
```
Comment (lalitmee) [inline .prettierrc:3]:
```suggestion
  "tabWidth": 2,
  "tabs": false,
```

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+	"extends": ["eslint:recommended", "plugin:prettier/recommended"]
```
Comment (lalitmee) [inline .eslintrc:2]: use this in the already present `.eslintrc.json`

**Review by lalitmee — CHANGES_REQUESTED** [review]

**Review by lalitmee — CHANGES_REQUESTED** [review]
also, add a proper `.prettierignore` file. Refer this [doc](https://prettier.io/docs/en/ignore.html)

**Comment by rohanmittal1163** [comment]
@lalitmee Done all the changes.
pls look into it 
thank you

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (lalitmee) [inline .prettierrc:1]: This file is using a big number of spaces. Please format it again with just `2` spaces indentation using `prettier`.

File: .prettierignore [hunk 7]
```diff
@@ -0,0 +1,5 @@
+*.git
+*.svn
+*.hg
+
+node_modules
```
Comment (lalitmee) [inline .prettierignore:5]: Is this working fine for `node_modules`? Can you try going into any file inside the `node_modules` and formatting it? If it formats it means it's not working and vice-versa.

File: .eslintrc.json [hunk 8]
```diff
@@ -1,3 +1,7 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": [
+    "eslint:recommended",
```
Comment (lalitmee) [inline .eslintrc.json:3]: please test these changes.

**Review by lalitmee — CHANGES_REQUESTED** [review]

File: .prettierignore (hunk 7 above)
Comment (rohanmittal1163) [reply .prettierignore:5]: yes node_modules are not formatting

File: .prettierrc (hunk 6 above)
Comment (rohanmittal1163) [reply .prettierrc:1]: Ok done

File: .eslintrc.json (hunk 8 above)
Comment (rohanmittal1163) [reply .eslintrc.json:3]: image
yes tested

**Comment by rohanmittal1163** [comment]
@lalitmee pls review to it.

File: package.json [hunk 9]
```diff
@@ -34,7 +33,9 @@
     "@types/node": "20.2.3",
     "@types/react": "18.2.6",
     "autoprefixer": "^10.4.14",
+    "eslint": "^8.41.0",
```
[the remark this answers is not in the document]
Comment (lalitmee) [reply package.json:36]: Why you didn't do this?

**Review by lalitmee — CHANGES_REQUESTED** [review]

**Comment by rohanmittal1163** [comment]
@lalitmee 
image
these are installed as devDependencies not the Dependencies to keep the production environment lean and avoids unnecessary bloat.

**Review by lalitmee — APPROVED** [review]

**Review by lalitmee — COMMENTED** [review]
@rohanmittal1163, can you do one thing? Run prettier on all of the project files and then push it.

@PiyushKalyanpy, I think we should do this here only. So that all the files are formatted.

**Comment by rohanmittal1163** [comment]
@lalitmee
Yeah sure ill do it

File: package-lock.json
Comment (PiyushKalyanpy) [inline package-lock.json:1]: remove this file from PR.

**Review by PiyushKalyanpy — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (rohanmittal1163) [reply package-lock.json:1]: Ohkk i ll change this and send new PR tomorrow itself

Comment (lalitmee) [reply package-lock.json:1]: @rohanmittal1163, why create a new PR? Let's just fix the `package-lock.json` and then in later PRs, we can run prettier on the codebase.

**Comment by rohanmittal1163** [comment]
@lalitmee i have format all the files
pls review the changes i made 
is there anything i  can do in this repo
thank you for your time

**Comment by PiyushKalyanpy** [comment]
Hello @rohanmittal1163 deleting file is not a solution.

**Comment by rohanmittal1163** [comment]
@PiyushKalyanpy as you have mentioned that remove this file from PR. 
is there something i understood wrong?

**Comment by PiyushKalyanpy** [comment]
Please create a new PR and make sure to fork the repo or update the repo with the new changes to avoid conflicts.

**Pull request closed without merging**